velox/type/Variant.h

```diff
@@ -411,7 +411,7 @@ class variant {
 
   variant(TypeKind kind) : kind_{kind}, ptr_{nullptr} {
     VELOX_CHECK(
-        !isDecimalKind(kind),
+        isDecimalKind(kind),
```

Comment:
If we remove the check for decimal kind, the check for other kind except decimal can work?

Reply:
This check is a positive check, and that means if the condition does not match, exception will be thrown. Here the logic was just reverted.

Reply:
See https://github.com/JkSelf/velox/blob/main/velox/common/base/Exceptions.h#L172-L182.